mws-simple.js

```diff
@@ -6,25 +6,26 @@ const getContentType = require('./lib/getContentType');
 
 const { name: pkgAppId, version: pkgAppVersionId } = require('./package.json'); // pkgAppId=name, pkgAppVersionId=version
 
+function promisify (data) {
+    return new Promise ((resolve, reject) => {
+        this.request(data, (err, result) => 
+            err ? reject(err) : resolve(result))
+    });
+}
+
 class MWSSimple {
     constructor({ appId = pkgAppId, appVersionId = pkgAppVersionId, host = 'mws.amazonservices.com', port = 443, accessKeyId, secretAccessKey, merchantId, authToken } = {}) {
         Object.assign(this, { appId, appVersionId, host, port, accessKeyId, secretAccessKey, merchantId, authToken, ServerError });
         
         // allows to use this inside the request method
-        this.request = this.request.bind(this);
+        promisify = promisify.bind(this);
```

Comment:
So, this fails pretty hard now, after this change.  (npm test fails the Promise check, as the server receives the wrong credentials) 

I think you could use node 8's 'utils' or 'util' lib that has a built-in promisify, and just make sure to do the this.request.bind(this) in constructor.  As is, putting promisify outside the class, doesn't work, as every time you construct a new class instance, it's changing the definition of promisify inside the module, to be bound to a different instance.

Reply:
If you have a MWS login, please put your keys in test/keys.json, and run npm test . make sure you don't commit that file, though. :-)  The test for Promise usage fails, as it checks the server return value, expecting "Not Found" but instead receives "Not Authorized", as 'this' is still pointing at the fake credentials made in the mws-simple defaults test